pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -517,6 +565,23 @@ func createRedirectedTaskSpec(kubeclient kubernetes.Interface, ts *v1alpha1.Task
 			EmptyDir: &corev1.EmptyDirVolumeSource{},
 		},
 	})
+
+	// Add the downward api volume
+	ts.Volumes = append(ts.Volumes, corev1.Volume{
+		Name: entrypoint.DownwardMountName,
+		VolumeSource: corev1.VolumeSource{
+			DownwardAPI: &corev1.DownwardAPIVolumeSource{
```

Comment:
i have a naiive question here about `DownwardAPIVolumeSource`, i tried to find it in the kube docs but failed: when this volume is mounted, is the content of the volume kept up to date, or is it snapshotted at mount time only? (and if it is kept up to date, how is that done? possibly relevant to our Pull Request resource plans... :innocent: )

Reply:
It's kept up to date, and afaik the kubelet handles keeping it up to date for each pod on the host by just watching the pod spec and updating appropriately.

Reply:
ah interesting, good to know!